include/clap/ext/draft/background-operation.h

```diff
@@ -0,0 +1,92 @@
+#pragma once
+
+#include "../../plugin.h"
+
+static const char CLAP_EXT_BACKGROUND_OPERATION[] = "clap.background_operation.draft/0";
+
+/*
+
+This extension let the host execute some operations on a background thread.
+
+Some operations can be CPU intensive and require a considerable amount of time, and being force to
+run them on the main thread prevent parallelisation.
+
+For example, loading a state or activating a plugin can require a bit of processing.
+When loading a large project, with a thousand of plugin instances, it is desirable to be able
+to load the plugin state and activate the plugin in a background thread. This can significantly
+improve the loading time.
+
+Concurrency isn't a trivial thing, and this extension should only be implemented if the plugin has
+expensive tasks to perform during load state or activate.
+
+Here is how this extension works:
+
+1. Check which operation can be performed in background
+
+        plugin_bgop->is_supported(plugin, CLAP_EXT_STATE);
+
+2. Inform the plugin that the host is about to start a background operation
+
+        const bool can_start = plugin_bgop->about_to_start(plugin);
+
+   Once about_to_start() is called, no more call can be made on the main thread.
```

Comment:
This is mentioned below, but I think it would help readability if we stressed here that no calls can be made on the main thread by either plug-in or host.